File: rails_api/app/controllers/benchmarks_controller.rb
# frozen_string_literal: true

# controller class for benchmarks
class BenchmarksController < ApplicationController
  def index
    i = 0
    i += 1 while i < 1_000_000
    sleep 0.02
    message = 'it works'
    render json: { message: message }
  end
end

File: phoenix_api/lib/phoenix_api_web/controllers/benchmarks_controller.ex
defmodule PhoenixApiWeb.BenchmarksController do
  use PhoenixApiWeb, :controller

  def index(conn, _params) do
    1..1000000 |> Enum.each( fn(a) ->  end)
    :timer.sleep(20)
    conn
    |> put_status(200)
    |> json(%{message: "it works!"})
  end
end
